### DISCUSSION TITLE
Way to split/organize the system definition?

### DISCUSSION THREAD

**Discussion by LeCalicot** [bevyengine/bevy#2781] [Q&A] [answered]
I'm pretty new to bevy and rust, so maybe I'm missing something obvious.

For the sake readability, I would like to gather some of the system calls together to reduce the number of calls chained together.

So, schematically, going from something like:
`App::build().add_system(my_system_1).add_system(my_system_2).add_system(my_system_3).run()`

To:
`App::build().call_sys_1_2_3().run()`
where the function call_sys_1_2_3 would be defined somewhere else.  I could then group system that are interacting together.

What's the easiest way to achieve that ? I tried defining function for that but I had trouble with the type of the argument and return.
I imagine it could be possible using a macro...?

**Comment by afonsolage** [comment, marked as answer]
Maybe you are looking for [Plugins](https://github.com/bevyengine/bevy/blob/main/examples/app/plugin.rs)?

**Reply by alice-i-cecile** [reply]
Plugins would be my first preference here. System sets also work quite well :)

**Reply by LeCalicot** [reply]
Ah. Oh. Indeed.

I will look into that but it seems to do what I had in mind.

**Reply by afonsolage** [reply]
If this is the correct answer, feel free to mark it as answered, so future folks can also check it :smiley:
